File: 4. Binary Tree/9_how_high.java
class Node<T> {
    T val;
    Node<T> left;
    Node<T> right;

    public Node(T val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

class Source {

    public static <T> int howHigh(Node<T> root) {
        if (root == null) {
            return -1;
        }
        return 1 + Math.max(howHigh(root.left), howHigh(root.right));
    }
}

File: 4. Binary Tree/9_how_high.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def how_high(root):
    if root is None:
        return -1
    return 1 + max(how_high(root.left), how_high(root.right))
